google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQuery.java

```diff
@@ -894,7 +898,54 @@ public int hashCode() {
    *
    * @throws BigQueryException upon failure
    */
-  Page<FieldValueList> listTableData(TableId tableId, TableDataListOption... options);
+  TableResult listTableData(TableId tableId, TableDataListOption... options);
+
+  /**
+   * Lists the table's rows. If the {@code schema} is not {@code null}, it is available to the
+   * {@link FieldValueList} iterated over.
+   *
+   * <p>Example of listing table rows.
+   *
+   * <pre>{@code
+   * String datasetName = "my_dataset_name";
+   * String tableName = "my_table_name";
+   * Schema schema = ...;
+   * String field = "my_field";
+   * Page<FieldValueList> tableData =
+   *     bigquery.listTableData(datasetName, tableName, schema);
+   * for (FieldValueList row : tableData.iterateAll()) {
+   *   row.get(field)
+   * }
+   * }</pre>
+   *
+   * @throws BigQueryException upon failure
+   */
+  TableResult listTableData(
+      String datasetId, String tableId, Schema schema, TableDataListOption... options);
+
+  /**
+   * Lists the table's rows. If the {@code schema} is not {@code null}, it is available to the
+   * {@link FieldValueList} iterated over.
+   *
+   * <p>Example of listing table rows.
+   *
+   * <pre>{@code
+   * Schema schema =
+   *     Schema.of(
+   *         Field.of("word", LegacySQLTypeName.STRING),
+   *         Field.of("word_count", LegacySQLTypeName.STRING),
+   *         Field.of("corpus", LegacySQLTypeName.STRING),
+   *         Field.of("corpus_date", LegacySQLTypeName.STRING));
+   * Page<FieldValueList> page =
+   *     bigquery.listTableData(
```

Comment:
Ditto. Should `page` be a `TableResult`?

Reply:
Done